src/libcore/slice.rs

```diff
@@ -1212,6 +1212,9 @@ impl<'a, T> Iterator for Windows<'a, T> {
     }
 }
 
+#[stable]
+impl<'a, T> ExactSizeIterator for Windows<'a, T> {}
```

Comment:
This implementation may not forever be right as the above `size_hint` may sometimes return `None` for the upper bound (albeit somewhat rarely)